File: tests/Test.DrawingCanvas.List.fs
module Test.DrawingCanvas.List

open Fable.Mocha
open Fable.React.DrawingCanvas
open Fable.React.DrawingCanvas.ListHelpers

let expectListEqual (expected:List<'T>) (value: List<'T>) =
    Expect.areEqual expected.Length value.Length
    for p in List.zip expected value do
        Expect.areEqual (fst p) (snd p)


let tests = testList "DrawingCanvas.List" [

    testCase "List ifThen true" <| fun () ->
        let d = [
            Save
            ifThen true (lazy [ BeginPath; Fill ])
            Restore
        ]

        let expected = [
            Save
            Insert [ BeginPath; Fill ]
            Restore
        ]

        expectListEqual expected d

    testCase "List IfThen false" <| fun () ->
        let d = [
            Save
            ifThen false (lazy [ Fill ])
            Restore
        ]

        let expected = [
            Save
            Insert []
            Restore
        ]

        expectListEqual expected d

    testCase "Builder IfThenElse true" <| fun () ->
        let d = [
            Save
            ifThenElse true (lazy [ BeginPath; Fill ]) (lazy [ BeginPath; Stroke ])
            Restore
        ]

        let expected = [
            Save
            Insert [ BeginPath; Fill ]
            Restore
        ]

        expectListEqual expected d

    testCase "List IfThenElse false" <| fun () ->
        let d = [
            Save
            ifThenElse false (lazy [ BeginPath; Fill ]) (lazy [ BeginPath; Stroke ])
            Restore
        ]

        let expected = [
            Save
            Insert [ BeginPath; Stroke ]
            Restore
        ]

        expectListEqual expected d

]

File: tests/Test.DrawingCanvas.List.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.tests = void 0;

var _DrawingCanvas = require("./Fable.React.DrawingCanvas/DrawingCanvas");

var _List = require("./fable-library.2.13.0/List");

var _Util = require("./Util");

var _Mocha = require("./Fable.Mocha.1.0.0/Mocha");

const tests = (0, _Mocha.Test$$$testList)("DrawingCanvas.List", (0, _List.ofArray)([(() => {
  return (0, _Mocha.Test$$$testCase)("List ifThen true", function () {
    const d = (0, _List.ofArray)([new _DrawingCanvas.CanvasCommand(1, "Save"), (() => {
      throw 1;
    })(), new _DrawingCanvas.CanvasCommand(2, "Restore")]);
    const expected = (0, _List.ofArray)([new _DrawingCanvas.CanvasCommand(1, "Save"), new _DrawingCanvas.CanvasCommand(3, "BeginPath"), new _DrawingCanvas.CanvasCommand(4, "Fill"), new _DrawingCanvas.CanvasCommand(2, "Restore")]);
    (0, _Util.expectDrawingsEqual)(expected, (() => {
      throw 1;
    })());
  });
})(), (() => {
  return (0, _Mocha.Test$$$testCase)("List IfThen false", function () {
    const d$$1 = (0, _List.ofArray)([new _DrawingCanvas.CanvasCommand(1, "Save"), (() => {
      throw 1;
    })(), new _DrawingCanvas.CanvasCommand(2, "Restore")]);
    const expected$$1 = (0, _List.ofArray)([new _DrawingCanvas.CanvasCommand(1, "Save"), new _DrawingCanvas.CanvasCommand(2, "Restore")]);
    (0, _Util.expectDrawingsEqual)(expected$$1, (() => {
      throw 1;
    })());
  });
})(), (() => {
  return (0, _Mocha.Test$$$testCase)("List IfThenElse true", function () {
    const d$$2 = (0, _List.ofArray)([new _DrawingCanvas.CanvasCommand(1, "Save"), (() => {
      throw 1;
    })(), new _DrawingCanvas.CanvasCommand(2, "Restore")]);
    const expected$$2 = (0, _List.ofArray)([new _DrawingCanvas.CanvasCommand(1, "Save"), new _DrawingCanvas.CanvasCommand(3, "BeginPath"), new _DrawingCanvas.CanvasCommand(4, "Fill"), new _DrawingCanvas.CanvasCommand(2, "Restore")]);
    (0, _Util.expectDrawingsEqual)(expected$$2, (() => {
      throw 1;
    })());
  });
})(), (() => {
  return (0, _Mocha.Test$$$testCase)("List IfThenElse false", function () {
    const d$$3 = (0, _List.ofArray)([new _DrawingCanvas.CanvasCommand(1, "Save"), (() => {
      throw 1;
    })(), new _DrawingCanvas.CanvasCommand(2, "Restore")]);
    const expected$$3 = (0, _List.ofArray)([new _DrawingCanvas.CanvasCommand(1, "Save"), new _DrawingCanvas.CanvasCommand(3, "BeginPath"), new _DrawingCanvas.CanvasCommand(5, "Stroke"), new _DrawingCanvas.CanvasCommand(2, "Restore")]);
    (0, _Util.expectDrawingsEqual)(expected$$3, (() => {
      throw 1;
    })());
  });
})()]));
exports.tests = tests;